traffic_ops/testing/api/v14/deliveryservices_required_capabilities_test.go

```diff
@@ -290,14 +297,14 @@ func DeleteTestDeliveryServicesRequiredCapabilities(t *testing.T) {
 	}
 }
 
-func helperGetDeliveryServiceID(t *testing.T, capability tc.DeliveryServicesRequiredCapability) *int {
+func helperGetDeliveryServiceID(t *testing.T, xmlID *string) *int {
 	t.Helper()
-	ds, _, err := TOSession.GetDeliveryServiceByXMLID(*capability.XMLID)
+	ds, _, err := TOSession.GetDeliveryServiceByXMLID(*xmlID)
```

Comment:
The way `xmlID` is getting passed in here it could be `nil` if defined that way in the test data. That's not valid (I'm pretty sure) for the database, but this is getting called prior to creation so you can't make the assumption that it conforms properly to the data model. Even if you could, though, I'd still always advise that pointers constructed from external data be checked against null before being dereferenced.

tl;dr pls check if `xmlID` is `nil` dereferencing, either at the top of this helper or before calling it